### ISSUE TITLE
Can't find expanded impls after invalidating caches

### ISSUE THREAD

**Issue by blueglyph** [intellij-rust/intellij-rust#9592]
## Environment

* **IntelliJ Rust plugin version:** 0.4.180.4932-222
* **Rust toolchain version:** 1.64.0 (a55dd71d5 2022-09-19) x86_64-pc-windows-msvc
* **IDE name and version:** IntelliJ IDEA 2022.2.3 Ultimate Edition (IU-222.4345.14)
* **Operating system:** Windows 10 10.0
* **Macro expansion:** enabled

## Problem description

I'm getting a lot of false positives because the analysis fails to recognize traits like Copy (but also Display, ...), even on primitive types like `u64` or `f64`.

![image](https://user-images.githubusercontent.com/31706195/197158914-d2747bce-f869-4d24-b87f-08e4593a52c3.png)

Sometimes the errors disappear after 2-3 minutes, sometimes not. Sometimes they disappear after an Invalidate, sometimes not.

I didn't have so many problems with earlier versions of the plugin, something must have changed, is there a separate, slower process that examines the code in background?

## Steps to reproduce

The example is taken from this crate (I'm also joining the project since I modified it - it's WIP so my version doesn't compile). It frequently happens in all projects though, which makes the plugin _very_ hard to use at the moment.

- https://github.com/dtolnay/dragonbox
- revision 830d0483bdb048d83b270dd4228e8cf4a31cefbf of master (the last one, the project is not maintained anymore)
- file src/to_chars.rs
- lines 27-29 in function `to_chars`

This particular file has 51 similar errors, mostly on native types.

[idea-logs-20221021-1105434832648817802330266.zip](https://github.com/intellij-rust/intellij-rust/files/9837493/idea-logs-20221021-1105434832648817802330266.zip)

[dragonbox-master.zip](https://github.com/intellij-rust/intellij-rust/files/9837508/dragonbox-master.zip)

**Comment by blueglyph** [comment]
This time it disappeared after an Invalidate Caches & clear file system cache, so you may not be able to reproduce it. This tends to appear after working a while on the code.

**Comment by blueglyph** [comment]
Curiously, I had two IDEs opened and I did an Invalidate Caches in another project, after which the problem reappeared in this one. I'm wondering if something is overridden between projects.

It wouldn't be the first time I see issues in IntelliJ when several projects are open at the same time.

**Comment by blueglyph** [comment]
I noticed something interesting.

After I did the **Invalidate Caches** (post above) and ignored the false positives in the first project, I saw that they started creeping into the 2nd project too. I had to do **Invalidate Caches** and set **Clear file system cache** 
=> the problem disappeared in both IDEs.

![image](https://user-images.githubusercontent.com/31706195/197209902-d40dc78b-75a6-46ef-81b2-14a5043b1c60.png)

Just to test, I then did a simple **Invalidate Caches** again, without any option selected
=> the problem reappeared in both IDEs.

I did a new **Invalidate Caches** and set **Clear file system cache** again
=> the problem disappeared again.

So it's quite repeatable. I'm waiting a few minutes each time, to make sure the very slow background task had time to process the project files.

**Comment by dima74** [comment]
Looks like it is caused by #8647

Steps to reproduce:

* Create empty project with code 
  ```rust
  fn func(x: i32) {
      consume(x);
      consume(x);
  }
  fn consume<T>(x: T) {}
  ```
* Invalidate caches and restart **with** "Clear file system cache" option
* Invalidate caches and restart **without** "Clear file system cache" option
